src/sentry/data_export/endpoints/data_export.py

```diff
@@ -45,21 +46,34 @@ def validate(self, data):
         # Discover Pre-processing
         if data["query_type"] == ExportQueryType.DISCOVER_STR:
             # coerce the fields into a list as needed
-            fields = query_info.get("field", [])
-            if not isinstance(fields, list):
-                fields = [fields]
+            equations = []
+
+            base_fields = query_info.get("field", [])
+            if not isinstance(base_fields, list):
+                base_fields = [base_fields]
+
+            fields = []
+            if self.context.get("has_arithmetic"):
```

Comment:
Instead of passing them all as `fields` when the flag is off, should this validate and error early if there are equations when it's not allowed?

Reply:
I personally prefer (as close as possible) that we try to run as if the new feature didn't exist. The exception being for features that might be gated separately than the whole, ie. if arithmetic wasn't eventually gonna merge with discover